lib/CheckStatusHook.ts

```diff
@@ -1,23 +1,24 @@
 import { useQuery, UseQueryOptions } from '@tanstack/react-query'
 import { ApiError } from 'next/dist/server/api-utils'
-import { CheckStatusReponse, CheckStatusRequestBody } from './StatusTypes'
+import { CheckStatusReponse, CheckStatusRequest } from './StatusTypes'
 
 export const fetchCheckStatus = async (
-  request: CheckStatusRequestBody,
+  checkStatusRequest: CheckStatusRequest,
   init?: RequestInit
 ): Promise<CheckStatusReponse | null> => {
-  const response = await fetch('/api/check-status', {
-    ...init,
-    method: 'POST',
-    body: JSON.stringify(request),
-  })
+  const query = Object.keys(checkStatusRequest)
+    .map(
+      (key) => `${key}=${checkStatusRequest[key as keyof CheckStatusRequest]}`
+    )
+    .join('&')
```

Comment:
Why not build this into a common function, and re-used it in the api and the test?

Reply:
I thought about this too and was on my list on thing to improve. Searching online to simple way of doing it and found an simpler solution with the usage of `URLSearchParams`.

```js
const obj = {
  page: 2,
  limit: 10,
  filter: 'js',
};

const result = '?' + new URLSearchParams(obj).toString();
console.log(result); // 👉️ "?page=2&limit=10&filter=js"
```

Reply:
It's kind of nice, and easy to re-write the `URLSearchParams` ourselves if it requires a whole package to be installed just for that.